### PR TITLE
refactor: lightbox

### PR DESCRIPTION
**Pull request by kyubisation**

### CODE REVIEW

File: src/angular/lightbox/lightbox-ref.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { OverlayRef } from '@angular/cdk/overlay';
+import { SbbDialogRef, _SbbDialogContainerBase } from '@sbb-esta/angular/dialog';
+import { Subject } from 'rxjs';
+import { filter } from 'rxjs/operators';
+
+// Counter for unique dialog ids.
+let uniqueId = 0;
+
+/**
+ * Reference to a dialog opened via the SbbDialog service.
```
Comment (jeripeierSBB) [inline src/angular/lightbox/lightbox-ref.ts:11]:
```suggestion
 * Reference to a dialog opened via the SbbLightbox service.
```

File: src/angular/lightbox/lightbox-ref.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  /**
+   * Not supported for lightbox.
+   * @docs-private
+   */
+  updatePosition(): this {
+    return this;
+  }
+
+  /** Not supported for lightbox. */
+  updateSize(): this {
```
Comment (jeripeierSBB) [inline src/angular/lightbox/lightbox-ref.ts:48]: mark as @docs-private?

File: src/angular/lightbox/lightbox.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
+The `SbbLightbox` service can be used to open modal dialogs with SBB styling and animations.
+
+```html
+<div sbbLightboxHeader>Hi {{ data.name }}</div>
```
Comment (jeripeierSBB) [inline src/angular/lightbox/lightbox.md:4]:
```suggestion
<div sbbLightboxTitle>Hi {{ data.name }}</div>
```

File: src/angular/lightbox/lightbox.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+<div sbbLightboxContent>
+  <div>
+    What's your favorite animal?
+    <sbb-form-field label="Animal">
+      <input type="text" [(ngModel)]="data.animal" cdkFocusInitial />
+    </sbb-form-field>
+  </div>
+</div>
+<div sbbLightboxFooter>
```
Comment (jeripeierSBB) [inline src/angular/lightbox/lightbox.md:13]:
```suggestion
<div sbbLightboxActions>
```

File: src/angular/lightbox/lightbox.scss [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    }
+    @include mq($from: desktop5k) {
+      font-size: pxToRem($sizeFontDefault5k);
+      padding: pxToRem(42) pxToRem(160) pxToRem(34);
+    }
+  }
+}
+
+.sbb-lightbox-title-close-button {
+  position: absolute;
```
Comment (jeripeierSBB) [inline src/angular/lightbox/lightbox.scss:101]: I think the space between the circle and the cross icon is not correct as designed

File: src/angular/lightbox/lightbox.scss [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  align-items: center;
+  // center is used for ie11 compatibility. TODO: Remove once we drop ie11 support.
+  justify-content: center;
+  justify-content: space-between;
+
+  // Explicitly set a box-sizing since people commonly set `border-box`
+  // on all elements which will break the height of the dialog actions.
```
Comment (jeripeierSBB) [inline src/angular/lightbox/lightbox.scss:140]:
```suggestion
  // on all elements which will break the height of the lightbox actions.
```

File: src/angular/lightbox/lightbox.scss [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+  }
+}
+
+.sbb-lightbox-title-close-button {
+  position: absolute;
+  right: pxToRem(20);
+  cursor: pointer;
+  line-height: 0;
+  color: $sbbColorStorm;
+
```
Comment (jeripeierSBB) [inline src/angular/lightbox/lightbox.scss:106]:
```suggestion
outline: 0

```

File: src/angular/lightbox/lightbox.ts [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    @Optional() @Inject(SBB_LIGHTBOX_DEFAULT_OPTIONS) defaultOptions: SbbLightboxConfig,
+    @Inject(SBB_LIGHTBOX_SCROLL_STRATEGY) scrollStrategy: any,
+    @Optional() @SkipSelf() parentDialog: SbbLightbox,
+    overlayContainer: OverlayContainer
+  ) {
+    super(
+      overlay,
+      injector,
+      defaultOptions,
+      parentDialog as any,
```
Comment (jeripeierSBB) [inline src/angular/lightbox/lightbox.ts:72]: any not any solution? ;)

File: src/components-examples/angular/lightbox/index.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import { LightboxTemplateExample } from './lightbox-template/lightbox-template-example';
+import { LightboxExample, LightboxExampleContent } from './lightbox/lightbox-example';
+
+export {
+  LightboxExample,
+  LightboxExampleContent,
+  LightboxComponentExample,
+  LightboxComponentExampleContent,
+  LightboxConfirmationExample,
+  LightboxWithConfirmationOnClose as LightboxContent4Example,
```
Comment (jeripeierSBB) [inline src/components-examples/angular/lightbox/index.ts:27]: Why is a alias export required here?

File: src/components-examples/angular/lightbox/lightbox-component/lightbox-component-example-content.html [hunk 10]
```diff
@@ -0,0 +1,96 @@
+<sbb-lightbox-title>
+  <strong><span class="cdk-visually-hidden">Train number </span> 5 527 </strong>
```
Comment (jeripeierSBB) [inline src/components-examples/angular/lightbox/lightbox-component/lightbox-component-example-content.html:2]:
```suggestion
  <strong><span class="cdk-visually-hidden">Train number </span> IC 5 527 </strong>
```

File: src/components-examples/angular/lightbox/lightbox-component/lightbox-component-example.ts [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    lightboxRef.afterClosed().subscribe((result) => {
+      console.log(`Lightbox result: ${result}`);
+    });
+  }
+}
+
+/** Lightbox with content loaded from. footer button bar */
+@Component({
+  selector: 'sbb-lightbox-component-example-content',
+  templateUrl: 'lightbox-component-example-content.html',
```
Comment (jeripeierSBB) [inline src/components-examples/angular/lightbox/lightbox-component/lightbox-component-example.ts:27]: I'd inline the template

File: src/components-examples/angular/lightbox/lightbox-component/lightbox-component-example.ts [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  openDialog() {
+    const lightboxRef = this.lightbox.open(LightboxComponentExampleContent);
+
+    lightboxRef.afterClosed().subscribe((result) => {
+      console.log(`Lightbox result: ${result}`);
+    });
+  }
+}
+
+/** Lightbox with content loaded from. footer button bar */
```
Comment (jeripeierSBB) [inline src/components-examples/angular/lightbox/lightbox-component/lightbox-component-example.ts:24]: Something seems weird with this comment

File: src/components-examples/angular/lightbox/lightbox-component/lightbox-component-example.ts [hunk 13]
```diff
@@ -0,0 +1,31 @@
+import { Component } from '@angular/core';
+import { SbbLightbox } from '@sbb-esta/angular/lightbox';
+
+/**
+ * @title Lightbox
```
Comment (jeripeierSBB) [inline src/components-examples/angular/lightbox/lightbox-component/lightbox-component-example.ts:5]: Better a more descriptive title

File: src/components-examples/angular/lightbox/lightbox-confirmation/lightbox-confirmation-example.ts [hunk 14]
```diff
@@ -0,0 +1,65 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { SbbLightbox, SbbLightboxRef } from '@sbb-esta/angular/lightbox';
+
+export interface LightboxData {
+  animal: string;
+  name: string;
+}
+
+/**
+ * @title Lightbox
```
Comment (jeripeierSBB) [inline src/components-examples/angular/lightbox/lightbox-confirmation/lightbox-confirmation-example.ts:10]: Better a more descriptive title

File: src/components-examples/angular/lightbox/lightbox-confirmation/lightbox-confirmation-example.ts [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+  }
+}
+
+/**
+ * Lightbox Alert with confirmation button
+ * all into one Lightbox using manualCloseAction Observable
+ */
+@Component({
+  selector: 'sbb-lightbox-with-confirmation-on-close',
+  templateUrl: 'lightbox-with-confirmation-on-close.html',
```
Comment (jeripeierSBB) [inline src/components-examples/angular/lightbox/lightbox-confirmation/lightbox-confirmation-example.ts:37]: inline template?

File: src/components-examples/angular/lightbox/lightbox/lightbox-example.html [hunk 16]
```diff
@@ -0,0 +1,11 @@
+<h4>Share date with the Lightbox</h4>
```
Comment (jeripeierSBB) [inline src/components-examples/angular/lightbox/lightbox/lightbox-example.html:1]:
```suggestion
<h4>Share data with the Lightbox</h4>
```

File: src/components-examples/angular/lightbox/lightbox/lightbox-example.ts [hunk 17]
```diff
@@ -0,0 +1,48 @@
+import { Component, Inject } from '@angular/core';
+import { SbbLightbox, SbbLightboxRef, SBB_LIGHTBOX_DATA } from '@sbb-esta/angular/lightbox';
+
+export interface LightboxData {
+  animal: string;
+  name: string;
+}
+
+/**
+ * @title Lightbox
```
Comment (jeripeierSBB) [inline src/components-examples/angular/lightbox/lightbox/lightbox-example.ts:10]: Better a more descriptive title

File: src/components-examples/angular/lightbox/lightbox/lightbox-example.ts [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+    lightboxRef.afterClosed().subscribe((result) => {
+      console.log('Lightbox sharing data was closed');
+      this.animal = result;
+    });
+  }
+}
+
+@Component({
+  selector: 'sbb-lightbox-example-content',
+  templateUrl: 'lightbox-example-content.html',
```
Comment (jeripeierSBB) [inline src/components-examples/angular/lightbox/lightbox/lightbox-example.ts:37]: Inline template?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Cool idea to inherit lightbox from dialog.

Generally I'd suggest to inline all templates which are no the primary one of an example. Otherwise it is not possible to see the html code in example view.

File: src/angular/lightbox/lightbox-content-directives.ts [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+
+  constructor(
+    // The lightbox title directive is always used in combination with a `SbbDialogRef`.
+    // tslint:disable-next-line: lightweight-tokens
+    @Optional() lightboxRef: SbbLightboxRef<any>,
+    elementRef: ElementRef<HTMLElement>,
+    lightbox: SbbLightbox,
+    changeDetectorRef: ChangeDetectorRef
+  ) {
+    super(lightboxRef, elementRef, lightbox as any, changeDetectorRef);
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/angular/lightbox/lightbox-content-directives.ts:103]: Unfortunately not easily. I will change it to ` as unknown as SbbDialog` as a slight improvement.

File: src/angular/lightbox/lightbox.scss (hunk 5 above)
Comment (kyubisation) [reply src/angular/lightbox/lightbox.scss:101]: In what way? According to sketch, this seems correct.

File: src/components-examples/angular/lightbox/lightbox-component/lightbox-component-example.ts (hunk 11 above)
Comment (kyubisation) [reply src/components-examples/angular/lightbox/lightbox-component/lightbox-component-example.ts:27]: I'd like to extend our showcase example component to allow displaying additional files in the future. And also avoid loading css files that do not exist.

File: src/angular/lightbox/lightbox.scss (hunk 5 above)
Comment (jeripeierSBB) [reply src/angular/lightbox/lightbox.scss:101]: your version: Bildschirmfoto 2021-07-20 um 08 27 16

sketch version: Bildschirmfoto 2021-07-20 um 08 27 36

Comment (kyubisation) [reply src/angular/lightbox/lightbox.scss:101]: Discussed with Daniel Schindler. Use circle-cross icon.

**Pull request merged**